### PR TITLE
perf(dsv4): repin MI355X vLLM AgentX to the 2026-09-20 ROCm nightly / 将 MI355X vLLM AgentX 重新固定至 2026-09-20 ROCm nightly

### PR DESCRIPTION
**Pull request by Fangzhou-Ai**
## Description

Repin the DSV4 MI355X vLLM AgentX recipe (`dsv4-fp4-mi355x-vllm-agentic-mtp`) to the newest published ROCm nightly, `vllm/vllm-openai-rocm:nightly-rocm100-d05da62e9ccdf8e342b15bf6785d83224cc165af` (digest `sha256:461e9c79…`, Docker Hub pushed `2026-09-20T06:32:10Z`, tag commit vllm-project/vllm@d05da62e), up from `vllm/vllm-openai-rocm:nightly-eed1f3d0c6043bd494424a22443ee198dd56f657` on `main` (pinned in #3071).

The original revision of this PR pinned the 2026-09-15 build `nightly-rocm100-cd10ed6f…`. That tag is now five nightlies old, so the pin moved forward to the current one. `main` has also been merged into the branch, which moved this PR's `perf-changelog.yaml` entry to the physical end of the file; `main`'s changelog bytes are preserved byte-for-byte.

### What the new nightly adds

`d05da62e` is 294 upstream commits ahead of `cd10ed6f` and 352 ahead of the CSA merge commit. Besides [vllm-project/vllm#51794](https://github.com/vllm-project/vllm/pull/51794), which enables CSA (Compressed Sparse Attention) multi-stream overlap for DeepSeek-V4 on ROCm so the two compressor `wkv_gate` GEMMs run on aux HIP streams instead of serially on the main stream during decode, the range brings in:

| Upstream PR | Change |
| --- | --- |
| [#57132](https://github.com/vllm-project/vllm/pull/57132) | Reverts the DeepSeek-V4 accuracy breakdown on ROCm (accuracy-relevant for the eval leg) |
| [#57465](https://github.com/vllm-project/vllm/pull/57465) | Fixes fused MoE expert distribution for DeepSeek V4 |
| [#56853](https://github.com/vllm-project/vllm/pull/56853) | Enables HCA dual-stream overlap for DeepSeek-V4 |
| [#50455](https://github.com/vllm-project/vllm/pull/50455) | Fixes the sparse-indexer logits collapse on gfx950/gfx942 |
| [#56743](https://github.com/vllm-project/vllm/pull/56743), [#57434](https://github.com/vllm-project/vllm/pull/57434) | gfx950 decode top-k optimization and ragged-metadata reuse across layers |
| [#55934](https://github.com/vllm-project/vllm/pull/55934) | triton + triton_kernels 3.8 MXFP4 MoE support for DeepSeek-V4 |
| [#56885](https://github.com/vllm-project/vllm/pull/56885) | Bumps AITER to v0.1.22.post1 |

No recipe fields other than `image` change. The search space, concurrency lists (`1, 4, 8, 16, 32, 40, 48, 56, 64` without DP attention and `64, 128, 192` with DP attention plus `vllm-router` 0.1.14), `dram-utilization: 0.60`, and the `VLLM_ROCM_USE_AITER_FUSION_SHARED_EXPERTS=0` workaround from #3071 are untouched. `benchmarks/single_node/agentic/dsv4_fp4_mi355x_vllm_mtp.sh` reads the image from `amd-master.yaml` and is unmodified.

Note for the reviewer: this range contains several AITER and MXFP4 MoE fixes ([#57426](https://github.com/vllm-project/vllm/pull/57426), [#57204](https://github.com/vllm-project/vllm/pull/57204), [#56568](https://github.com/vllm-project/vllm/pull/56568)) that touch the shared-expert fusion path the `VLLM_ROCM_USE_AITER_FUSION_SHARED_EXPERTS=0` workaround guards against. The workaround is deliberately left in place here because nothing in this PR establishes that it is safe to drop; re-testing it is a separate change.

## AI model disclosure

- Model/version: `claude-opus-5-thinking-high` as exposed by the Cursor runtime. The exact upstream build string is not exposed, so it could not be verified beyond this identifier.
- Role: Queried Docker Hub for the newest published `nightly-rocm100` tag and its digest, verified upstream commit ancestry via the GitHub API, updated `configs/amd-master.yaml` and the `perf-changelog.yaml` entry, merged `main` and resolved the changelog conflict with `infx.workflows.prepare_perf_changelog_merge`, ran the local validation below, and wrote this description. No work was delegated to other agents or models.
- Earlier revisions of this PR were prepared with a model the current runtime cannot identify; that disclosure could not be verified retroactively.

## Type of Change

- [x] Configuration change

## Validation

Local, on head commit `10adbcd93`:

- `python3 -m infx.workflows.validate_perf_changelog --base-ref origin/main --head-ref HEAD` → `Validated perf-changelog.yaml: final newline present and matrix generated`.
- `infx.matrix.generate test-config --config-keys dsv4-fp4-mi355x-vllm-agentic-mtp` emits 12 rows, and every row's `image` is `vllm/vllm-openai-rocm:nightly-rocm100-d05da62e9ccdf8e342b15bf6785d83224cc165af`. Inspected fields are unchanged versus the pre-bump generation: `tp: 8`, `pp: 1`, `ep: 1`, `dcp-size: 1`, `pcp-size: 1`, `spec-decoding: mtp`, `scenario-type: agentic-coding`, `kv-offloading: none`, `run-eval: false`, concurrency `1/4/8/16/32/40/48/56/64` at `dp-attn: false` and `64/128/192` at `dp-attn: true` with `router: {vllm-router, 0.1.14}`.
- `pytest utils/test_process_changelog.py utils/changelog_gate_tests/` → 139 passed.
- YAML parse of `configs/amd-master.yaml`, `configs/runners.yaml`, and `perf-changelog.yaml`; `git diff --check`; `bash -n benchmarks/single_node/agentic/dsv4_fp4_mi355x_vllm_mtp.sh`.
- Byte check: the first `N` bytes of `perf-changelog.yaml` are identical to `origin/main:perf-changelog.yaml` (`cmp` clean), with this PR's single entry appended after them.
- Registry check: the tag and digest were read from the Docker Hub v2 API; `nightly-rocm100-d05da62e…` is the most recently pushed `nightly-rocm100` commit tag.
- Ancestry check: `GET /repos/vllm-project/vllm/compare/cd10ed6f…...d05da62e…` reports `status: ahead, ahead_by: 294, behind_by: 0`, and the [#51794](https://github.com/vllm-project/vllm/pull/51794) merge commit `a6c5d6d0fcd7dd5b49e3146991c201b16caebef2` is 352 commits behind `d05da62e`, so it is included.

**Not yet proven:** no GPU run has executed this image. Local generation and registry metadata are not runtime evidence. The `full-sweep-enabled` PR sweep on this head commit is the gate for MI355X throughput and eval results.

<details>
<summary>中文</summary>

## 说明

将 DSV4 MI355X vLLM AgentX 配方（`dsv4-fp4-mi355x-vllm-agentic-mtp`）重新固定到最新发布的 ROCm nightly：`vllm/vllm-openai-rocm:nightly-rocm100-d05da62e9ccdf8e342b15bf6785d83224cc165af`（digest `sha256:461e9c79…`，Docker Hub 推送于 `2026-09-20T06:32:10Z`，标签提交 vllm-project/vllm@d05da62e），替换 `main` 上由 #3071 固定的 `vllm/vllm-openai-rocm:nightly-eed1f3d0c6043bd494424a22443ee198dd56f657`。

本 PR 最初固定的是 2026-09-15 的构建 `nightly-rocm100-cd10ed6f…`，该标签目前已落后五个 nightly，因此将固定版本推进到当前最新版本。同时已将 `main` 合并进本分支，使本 PR 的 `perf-changelog.yaml` 条目移动到文件物理末尾；`main` 的 changelog 字节完整保留、逐字节一致。

### 新 nightly 带来的改动

`d05da62e` 比 `cd10ed6f` 前进 294 个上游提交，比 CSA 合并提交前进 352 个。除 [vllm-project/vllm#51794](https://github.com/vllm-project/vllm/pull/51794)（为 ROCm 上的 DeepSeek-V4 启用 CSA（压缩稀疏注意力）多流重叠，使两个 compressor `wkv_gate` GEMM 在解码时运行于辅助 HIP 流，而非在主流上串行执行）之外，该区间还包含：

| 上游 PR | 改动 |
| --- | --- |
| [#57132](https://github.com/vllm-project/vllm/pull/57132) | 回退导致 ROCm 上 DeepSeek-V4 精度崩塌的改动（与 eval 环节的精度相关） |
| [#57465](https://github.com/vllm-project/vllm/pull/57465) | 修复 DeepSeek V4 的融合 MoE 专家分布 |
| [#56853](https://github.com/vllm-project/vllm/pull/56853) | 为 DeepSeek-V4 启用 HCA 双流重叠 |
| [#50455](https://github.com/vllm-project/vllm/pull/50455) | 修复 gfx950/gfx942 上稀疏 indexer 的 logits 崩塌 |
| [#56743](https://github.com/vllm-project/vllm/pull/56743)、[#57434](https://github.com/vllm-project/vllm/pull/57434) | gfx950 解码 top-k 优化，以及跨层复用 ragged 元数据 |
| [#55934](https://github.com/vllm-project/vllm/pull/55934) | 为 DeepSeek-V4 提供 triton + triton_kernels 3.8 MXFP4 MoE 支持 |
| [#56885](https://github.com/vllm-project/vllm/pull/56885) | 将 AITER 升级至 v0.1.22.post1 |

除 `image` 外未改动任何配方字段。搜索空间、并发列表（不启用 DP attention 时为 `1, 4, 8, 16, 32, 40, 48, 56, 64`，启用 DP attention 时为 `64, 128, 192` 并搭配 `vllm-router` 0.1.14）、`dram-utilization: 0.60`，以及 #3071 引入的 `VLLM_ROCM_USE_AITER_FUSION_SHARED_EXPERTS=0` 规避方案均保持不变。`benchmarks/single_node/agentic/dsv4_fp4_mi355x_vllm_mtp.sh` 从 `amd-master.yaml` 读取镜像，未做修改。

提请审阅者注意：该提交区间包含若干 AITER 与 MXFP4 MoE 修复（[#57426](https://github.com/vllm-project/vllm/pull/57426)、[#57204](https://github.com/vllm-project/vllm/pull/57204)、[#56568](https://github.com/vllm-project/vllm/pull/56568)），涉及 `VLLM_ROCM_USE_AITER_FUSION_SHARED_EXPERTS=0` 所规避的 shared-expert fusion 路径。本 PR 有意保留该规避方案，因为本 PR 无法证明现在可以安全移除；重新验证该项应作为单独的改动处理。

## AI 模型使用说明

- 模型/版本：Cursor 运行环境提供的标识为 `claude-opus-5-thinking-high`。运行环境未提供确切的上游构建字符串，因此除该标识外无法进一步确认。
- 工作内容：查询 Docker Hub 获取最新发布的 `nightly-rocm100` 标签及其 digest，通过 GitHub API 校验上游提交祖先关系，更新 `configs/amd-master.yaml` 与 `perf-changelog.yaml` 条目，合并 `main` 并使用 `infx.workflows.prepare_perf_changelog_merge` 解决 changelog 冲突，执行下述本地校验，并撰写本说明。未将任何工作委派给其他 agent 或模型。
- 本 PR 早前的修订由当前运行环境无法识别的模型完成，该部分说明无法追溯确认。

## 改动类型

- [x] 配置变更

## 验证

在 head 提交 `10adbcd93` 上本地执行：

- `python3 -m infx.workflows.validate_perf_changelog --base-ref origin/main --head-ref HEAD` → `Validated perf-changelog.yaml: final newline present and matrix generated`。
- `infx.matrix.generate test-config --config-keys dsv4-fp4-mi355x-vllm-agentic-mtp` 生成 12 行，每行的 `image` 均为 `vllm/vllm-openai-rocm:nightly-rocm100-d05da62e9ccdf8e342b15bf6785d83224cc165af`。已检查字段与升级前的生成结果一致：`tp: 8`、`pp: 1`、`ep: 1`、`dcp-size: 1`、`pcp-size: 1`、`spec-decoding: mtp`、`scenario-type: agentic-coding`、`kv-offloading: none`、`run-eval: false`，`dp-attn: false` 时并发为 `1/4/8/16/32/40/48/56/64`，`dp-attn: true` 时并发为 `64/128/192` 并带 `router: {vllm-router, 0.1.14}`。
- `pytest utils/test_process_changelog.py utils/changelog_gate_tests/` → 139 通过。
- 已完成 `configs/amd-master.yaml`、`configs/runners.yaml`、`perf-changelog.yaml` 的 YAML 解析；`git diff --check`；`bash -n benchmarks/single_node/agentic/dsv4_fp4_mi355x_vllm_mtp.sh`。
- 字节检查：`perf-changelog.yaml` 的前 `N` 字节与 `origin/main:perf-changelog.yaml` 完全一致（`cmp` 无差异），本 PR 的单个条目追加在其后。
- 镜像仓库检查：标签与 digest 来自 Docker Hub v2 API；`nightly-rocm100-d05da62e…` 是最近推送的 `nightly-rocm100` 提交标签。
- 祖先关系检查：`GET /repos/vllm-project/vllm/compare/cd10ed6f…...d05da62e…` 返回 `status: ahead, ahead_by: 294, behind_by: 0`；[#51794](https://github.com/vllm-project/vllm/pull/51794) 的合并提交 `a6c5d6d0fcd7dd5b49e3146991c201b16caebef2` 比 `d05da62e` 落后 352 个提交，因此已包含在内。

**尚未证明：** 还没有任何 GPU 运行使用过该镜像。本地生成与镜像仓库元数据不构成运行时证据。本 head 提交上的 `full-sweep-enabled` PR sweep 才是 MI355X 吞吐与 eval 结果的验证门禁。

</details>

### CODE REVIEW

File: configs/amd-master.yaml [hunk 1]
```diff
@@ -1310,7 +1310,7 @@ qwen3.5-fp8-mi325x-sglang-mtp:
       - { tp: 8, ep: 1, conc-start: 4, conc-end: 64, spec-decoding: mtp }
 
 dsv4-fp4-mi355x-vllm-agentic-mtp:
-  image: vllm/vllm-openai-rocm:nightly-eed1f3d0c6043bd494424a22443ee198dd56f657
+  image: vllm/vllm-openai-rocm:nightly-rocm100-cd10ed6f9f6b37a8ace9cf380007e66fe12ec0c3
```
[the remark this answers is not in the document]
Comment (Fangzhou-Ai) [reply configs/amd-master.yaml:1313]: Stale — anchored to commit 16ed0aec (the image-bump-only commit) before the changelog entry was added. The very next commit, [02f50f86](https://github.com/SemiAnalysisAI/InferenceX/commit/02f50f8603f1bd44f05bf1c64800ba1d4abd37e9), appends the required `perf-changelog.yaml` entry for `dsv4-fp4-mi355x-vllm-agentic-mtp`, and the PR description's `PENDING_CHANGELOG_NOTE` placeholder was replaced before this comment posted. Resolving.
